### ISSUE TITLE
MongoRocks does not return documents when reverse ordering is needed.

### ISSUE THREAD

**Issue by SunguckLee** [mongodb-partners/mongo-rocks#144]
MongoRocks does not return documents when reverse ordering is needed.
  MongoDB : 3.4.14
  MongoRocks : 3.4.9
  RocksDB : 5.12

Weired thing is that reverse order find return documents after restarting mongodb.
And this case does not happen on mongodb 3.4.14 + rocksdb 5.11 (or rocksdb 5.9).

This is simple reproducing scenario.
```
> db.notfound.createIndex({aver:1})
{
	"createdCollectionAutomatically" : true,
	"numIndexesBefore" : 1,
	"numIndexesAfter" : 2,
	"ok" : 1
}
> for(var idx=0; idx<5000000; idx++){
...   db.notfound.insert({aver:"20180418-1", fd1:idx, fd2:"20180321080310"+idx});
... }
WriteResult({ "nInserted" : 1 })

> db.notfound.count()
5000000

> db.notfound.find().sort({$natural:1})
{ "_id" : ObjectId("5b15e74c78b3645a3d411cf1"), "aver" : "20180418-1", "fd1" : 0, "fd2" : "201803210803100" }
{ "_id" : ObjectId("5b15e74c78b3645a3d411cf2"), "aver" : "20180418-1", "fd1" : 1, "fd2" : "201803210803101" }
{ "_id" : ObjectId("5b15e74c78b3645a3d411cf3"), "aver" : "20180418-1", "fd1" : 2, "fd2" : "201803210803102" }
{ "_id" : ObjectId("5b15e74c78b3645a3d411cf4"), "aver" : "20180418-1", "fd1" : 3, "fd2" : "201803210803103" }
{ "_id" : ObjectId("5b15e74c78b3645a3d411cf5"), "aver" : "20180418-1", "fd1" : 4, "fd2" : "201803210803104" }
{ "_id" : ObjectId("5b15e74c78b3645a3d411cf6"), "aver" : "20180418-1", "fd1" : 5, "fd2" : "201803210803105" }
{ "_id" : ObjectId("5b15e74c78b3645a3d411cf7"), "aver" : "20180418-1", "fd1" : 6, "fd2" : "201803210803106" }
{ "_id" : ObjectId("5b15e74c78b3645a3d411cf8"), "aver" : "20180418-1", "fd1" : 7, "fd2" : "201803210803107" }
{ "_id" : ObjectId("5b15e74c78b3645a3d411cf9"), "aver" : "20180418-1", "fd1" : 8, "fd2" : "201803210803108" }
{ "_id" : ObjectId("5b15e74c78b3645a3d411cfa"), "aver" : "20180418-1", "fd1" : 9, "fd2" : "201803210803109" }
{ "_id" : ObjectId("5b15e74c78b3645a3d411cfb"), "aver" : "20180418-1", "fd1" : 10, "fd2" : "2018032108031010" }
{ "_id" : ObjectId("5b15e74c78b3645a3d411cfc"), "aver" : "20180418-1", "fd1" : 11, "fd2" : "2018032108031011" }
{ "_id" : ObjectId("5b15e74c78b3645a3d411cfd"), "aver" : "20180418-1", "fd1" : 12, "fd2" : "2018032108031012" }
{ "_id" : ObjectId("5b15e74c78b3645a3d411cfe"), "aver" : "20180418-1", "fd1" : 13, "fd2" : "2018032108031013" }
{ "_id" : ObjectId("5b15e74c78b3645a3d411cff"), "aver" : "20180418-1", "fd1" : 14, "fd2" : "2018032108031014" }
{ "_id" : ObjectId("5b15e74c78b3645a3d411d00"), "aver" : "20180418-1", "fd1" : 15, "fd2" : "2018032108031015" }
{ "_id" : ObjectId("5b15e74c78b3645a3d411d01"), "aver" : "20180418-1", "fd1" : 16, "fd2" : "2018032108031016" }
{ "_id" : ObjectId("5b15e74c78b3645a3d411d02"), "aver" : "20180418-1", "fd1" : 17, "fd2" : "2018032108031017" }
{ "_id" : ObjectId("5b15e74c78b3645a3d411d03"), "aver" : "20180418-1", "fd1" : 18, "fd2" : "2018032108031018" }
{ "_id" : ObjectId("5b15e74c78b3645a3d411d04"), "aver" : "20180418-1", "fd1" : 19, "fd2" : "2018032108031019" }
Type "it" for more

> db.notfound.find().sort({$natural:-1})
==> Not Found
```

**Comment by SunguckLee** [comment]
Looks like this issue is resolved from RocksDB 5.13.4 (https://github.com/facebook/rocksdb/releases/tag/v5.13.4).

**Comment by igorcanadi** [comment]
Cool, glad that it's a bug in RocksDB, I was not looking forward to debugging this :)

Thanks @SunguckLee
